examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImWriteCommand.java

```diff
@@ -0,0 +1,79 @@
+package com.matter.controller.commands.pairing;
+
+import chip.devicecontroller.ChipDeviceController;
+import chip.devicecontroller.GetConnectedDeviceCallbackJni.GetConnectedDeviceCallback;
+import chip.devicecontroller.WriteAttributesCallback;
+import chip.devicecontroller.model.AttributeWriteRequest;
+import chip.devicecontroller.model.ChipAttributePath;
+import com.matter.controller.commands.common.CredentialsIssuer;
+import java.util.ArrayList;
+import java.util.logging.Level;
+
+public final class PairOnNetworkLongImWriteCommand extends PairingCommand
+    implements WriteAttributesCallback, GetConnectedDeviceCallback {
+  private static final int MATTER_PORT = 5540;
+  private long devicePointer;
+
+  public PairOnNetworkLongImWriteCommand(
+      ChipDeviceController controller, CredentialsIssuer credsIssue) {
+    super(
+        controller,
+        "onnetwork-long-im-write",
+        PairingModeType.ON_NETWORK,
+        PairingNetworkType.NONE,
+        credsIssue,
+        DiscoveryFilterType.LONG_DISCRIMINATOR);
+  }
+
+  @Override
+  protected void runCommand() {
+    // boolean true for tlv
+    byte[] booleanTLV = {0x09};
+    AttributeWriteRequest attribute = AttributeWriteRequest.newInstance(0, 0x0028, 16, booleanTLV);
+    ArrayList<AttributeWriteRequest> attributeList = new ArrayList<>();
+    attributeList.add(attribute);
+
+    currentCommissioner()
+        .pairDeviceWithAddress(
+            getNodeId(),
+            getRemoteAddr().getHostAddress(),
+            MATTER_PORT,
+            getDiscriminator(),
+            getSetupPINCode(),
+            null);
+    currentCommissioner().setCompletionListener(this);
+    waitCompleteMs(getTimeoutMillis());
+    currentCommissioner().getConnectedDevicePointer(getNodeId(), this);
+    clear();
+
+    currentCommissioner().write(this, devicePointer, attributeList, 0, 0);
+
+    waitCompleteMs(getTimeoutMillis());
+  }
+
+  public void onError(ChipAttributePath attributePath, Exception e) {
```

Comment:
Instead of making these interface methods public on the command (which makes them invokable externally rather than just as callbacks) consider defining inner classes for these:

```
public final class PairOnNetworkLongImWriteCommand extends PairingCommand {
  // ...

  currentCommissioner().write(new InternalWriteAttributesCallback(), devicePointer, attributeList, 0, 0);

  private class InternalWriteAttributesCallback implements WriteAttributesCallback {
    @Override
    public void onError(...) { ... }

    @Override
    public void onResponse(...) { ... }
  }
}
```

Reply:
good idea, updated, thanks